### ISSUE TITLE
Clicking a breadcrumbs link should focus on the H1 of the page

### ISSUE THREAD

**Issue by randimays** [department-of-veterans-affairs/vets-design-system-documentation#4820]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

We noticed that when you click a link in the breadcrumbs that is for the current page, the focus on the H1 on that page is lost. This should be maintained per [the latest accessibility guidance](https://design.va.gov/templates/forms/accessibility-guidelines#managing-focus-in-form-flows).

## What I expected to happen

When I click a link in the breadcrumbs that links to the same page I'm already looking at, it should maintain focus on (or re-throw focus to) the H1 on that page. I don't think this necessarily makes sense to implement on an app-by-app level because:

1) it is standard a11y guidance for the whole site (focus the H1)
2) breadcrumbs are also used on static pages without corresponding vets-website code and this would cover those
3) it is simpler to implement this solution in the source code of the breadcrumbs where there is easy access to the click events of the links

I did try to implement this for our app specifically, however. I say there there is an `onRouteChange` handler that can be passed to `<VaBreadcrumbs />` but that only fires when you click on links for pages **other than** the one you're currently looking at. I tried adding an event listener on the last breadcrumb of the trail, but that didn't work for me because we dynamically create that last breadcrumb (as it matches the H1 on the page its on) and I think there was a race condition, so it was always targeting the second-to-last breadcrumb.

## Reproducing

This can be reproduced on likely any, or at least most, pages on VA.gov such as https://www.va.gov/disability. Click on the "VA disability compensation" link in the breadcrumbs and note that, while you are navigated to the H1 / anchor of the page, if you have a screenreader / VoiceOver activated, you will not hear the heading read out and the yellow focus outline is not present.

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [x] This bug is not blocking any work
- [ ] Other

### Details

This bug is not blocking anything, but may come up in Staging Reviews as a blocker for new applications (such as ours - Decision Reviews Onramp tool) as a focus management issue.
